From 3adba4b9e36f5eb3c2611bef1dfce70f7db10de7 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@kde.org>
Date: Sun, 1 Mar 2026 14:38:46 +0100
Subject: [PATCH] plugins/gamecontroller: fix build on 32 bit architectures

The evdev event API is quite weird there, the input_event struct has different
contents depending on the architecture.

BUG: 516886
---
 src/plugins/gamecontroller/emulatedinputdevice.cpp | 4 ++--
 src/plugins/gamecontroller/gamecontroller.cpp      | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/plugins/gamecontroller/emulatedinputdevice.cpp b/src/plugins/gamecontroller/emulatedinputdevice.cpp
index e85fc1a513a..94859fa3653 100644
--- a/src/plugins/gamecontroller/emulatedinputdevice.cpp
+++ b/src/plugins/gamecontroller/emulatedinputdevice.cpp
@@ -74,7 +74,7 @@ void EmulatedInputDevice::emulateInputDevice(input_event &ev)
 void EmulatedInputDevice::evkeyMapping(input_event *ev)
 {
     KeyboardKeyState state = ev->value ? KWin::KeyboardKeyState::Pressed : KWin::KeyboardKeyState::Released;
-    const std::chrono::microseconds time = std::chrono::seconds(ev->time.tv_sec) + std::chrono::microseconds(ev->time.tv_usec);
+    const std::chrono::microseconds time = std::chrono::seconds(ev->input_event_sec) + std::chrono::microseconds(ev->input_event_usec);
 
     switch (ev->code) {
     case BTN_SOUTH: // A button → Enter
@@ -116,7 +116,7 @@ void EmulatedInputDevice::evabsMapping(input_event *ev)
     const int maximumValue = libevdev_get_abs_maximum(m_device, ev->code);
     const PointerButtonState pointerState = maximumValue > 0 && ev->value >= maximumValue * 0.9 ? KWin::PointerButtonState::Pressed : KWin::PointerButtonState::Released;
     KeyboardKeyState keyState = ev->value ? KWin::KeyboardKeyState::Pressed : KWin::KeyboardKeyState::Released;
-    const std::chrono::microseconds time = std::chrono::seconds(ev->time.tv_sec) + std::chrono::microseconds(ev->time.tv_usec);
+    const std::chrono::microseconds time = std::chrono::seconds(ev->input_event_sec) + std::chrono::microseconds(ev->input_event_usec);
 
     switch (ev->code) {
     // analog triggers
diff --git a/src/plugins/gamecontroller/gamecontroller.cpp b/src/plugins/gamecontroller/gamecontroller.cpp
index eae6f8f04bb..7d32cf4924e 100644
--- a/src/plugins/gamecontroller/gamecontroller.cpp
+++ b/src/plugins/gamecontroller/gamecontroller.cpp
@@ -137,7 +137,7 @@ void GameController::logEvent(input_event *ev)
         return;
     }
 
-    const qint64 timestamp = ev->time.tv_sec * 1000 + ev->time.tv_usec / 1000;
+    const qint64 timestamp = ev->input_event_sec * 1000 + ev->input_event_usec / 1000;
 
     if (ev->type == EV_SYN) {
         qCDebug(KWIN_GAMECONTROLLER) << "Device:" << libevdev_get_name(m_evdev.get()) << "time(ms):" << timestamp << "Event: -------------- SYN_REPORT ------------";
-- 
GitLab

